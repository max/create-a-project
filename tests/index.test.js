const path = require("path");
const ProjectCreator = require("..");

const FIXTURE_WORKSPACE = path.join(__dirname, "fixtures");

describe("create-a-project", () => {
  let projectCreator, github;

  beforeEach(() => {
    projectCreator = new ProjectCreator();
    github = { projects: { createForRepo: jest.fn() } };

    projectCreator.tools.workspace = FIXTURE_WORKSPACE;
    projectCreator.tools.context.payload = {
      repository: { owner: { login: "max" }, name: "waddup" }
    };
    projectCreator.tools.createOctokit = jest.fn(() => github);
  });

  it("creates a new project", async () => {
    await projectCreator.go();
    expect(github.projects.createForRepo).toHaveBeenCalled();
    expect(github.projects.createForRepo.mock.calls).toMatchSnapshot();
  });

  it("creates a new project from a different template", async () => {
    projectCreator = new ProjectCreator(".github/different-template.taskpaper");
    projectCreator.tools.workspace = FIXTURE_WORKSPACE;
    projectCreator.tools.context.payload = {
      repository: { owner: { login: "max" }, name: "waddup" }
    };
    projectCreator.tools.createOctokit = jest.fn(() => github);

    await projectCreator.go();
    expect(github.projects.createForRepo).toHaveBeenCalled();
    expect(github.projects.createForRepo.mock.calls[0][0].name).toBe(
      "My Different Fantastic Project Board"
    );
  });
});
