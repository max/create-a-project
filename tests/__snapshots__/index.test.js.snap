// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-a-project creates a new project 1`] = `
Array [
  Array [
    Object {
      "body": "My amazing project board description.",
      "name": "My Fantastic Project Board",
      "owner": "max",
      "repo": "waddup",
    },
  ],
]
`;
